## This is a configuration file for the different parameters used in the project

###### Filter parameters ######
## Voxel downsampling parameter -> float: for sphere extraction
voxel_down_sample: 0.75
## Sphere radius -> int: for sphere extraction
sphere_radius: 10
## Filtering radius -> float: for outlier removal
filtering_radius: 0.2
## Number of neighbors -> int: for outlier removal
nb_neighbors: 10
## Std ratio -> float: for outlier removal
std_ratio: 0.5

###### GICP parameters ######
## Maximum correspondence distance, step 1 -> float
max_cor_dist_1: 3.0
## Maximum correspondence distance, step 2 -> float
max_cor_dist_2: 1.0

###### Topics and Frames ######
## Frame ID -> str: for visualization
frame_id: 'map'
## Robot 1 map topic -> str
r1_topic: '/lio_sam/mapping/map_global'
## Robot 2 map topic -> str
r2_topic: '/lio_sam/mapping/map_local'
## Merged map topic -> str
merged_topic: '/merged_map'

###### Namespaces ######
## Robot 1 namespace -> str
r1_ns: 'robot1'
## Robot 2 namespace -> str
r2_ns: 'robot2'

###### Node parameters ######
## Number of threads -> int: for parallel processing
num_threads: 8
## Rate -> int: for node frequency (Hz)
rate: 1